### DISCUSSION TITLE
Easy Editor for Yamaha DX7

### DISCUSSION THREAD

**Discussion by mtarenskeen** [RomanKubiak/ctrlr#651] [Releases]
I am releasing a new editor panel for the Yamaha DX7 with a different approach than existing DX7 editing software. My goal was not to support each and every parameter in the DX7. My goal was to be able to quickly modify existing patches and create totally new sounds using a minimal number of controls. Creating new sounds, without the need to fully understand complex FM synthesis, has never been easier. Should also work with every DX7 compatible synth. For VolcaFM compatibility there is a special button, because Korg replaced the VCED checksum byte with an operator on/off mask for the six operators, breaking fulll compatibility with the DX7. I don't own a VolcaFM, so it would be nice if some VolcaFM owner could test this feature.

You can find the panel at https://ctrlr.martintarenskeen.nl

Bugreports are welcome. I would be surprised if there aren't any bugs ;-)

![ctrlr-dx7-easyedit](https://github.com/RomanKubiak/ctrlr/assets/3509304/978432cf-d925-4f89-aca3-c35d7316bd05)

**Comment by bijlevel** [comment]
I did make in the past a full blown editor for the DX-7 and TX-7, still available on the legacy CTRLR-website and rather complicated to work with. So I like your idea! Can it work with Dexed? BTW your link does not work from within Github and when copying the link my webbrowser does not want to show your website, thinks it is unsave...

**Comment by mtarenskeen** [comment]
If Dexed can receive VCED sysex dump data (single patch temporary editbuffer) it should work. I will try.

I changed the way the link is inserted in the GitHub message editor. Now I just typed the address without additional markup that was inserted with the paperclip icon. I think the link works now: https://ctrlr.martintarenskeen.nl

**Reply by bijlevel** [reply]
The link is now OK, thanks! I did have  a TX-7, but sold it, so that's why I asked about Dexed, because it is a good software emulation of a DX-7. But I still have to investigate if the plugin works in my DAW on my new Mac Studio. Thanks anyway for trying out Dexed!

**Reply by maxcomperatore** [reply]
Yes, Dexed accepts single-voice VCED (155-byte edit buffer) dumps over MIDI In on Channel 1! 

If you are testing your panel against physical DX7 hardware too, keep in mind the two classic firmware quirks that trip people up with VCED:

1. SYS INFO AVAIL: On original Mark I units, you have to press `FUNCTION` -> Button 8 (`MIDI`) and toggle `SYS INFO ?` to `SYS INFO AVAIL`. If left on `UNAVAIL` (the default on reboot), the synth silently discards incoming VCED messages without updating the edit buffer.
2. Channel 1 lock: Stock Mark I ROMs only listen for VCED and VMEM on MIDI Channel 1 (Yamaha sub-status byte `00` in `F0 43 00 00 ...`), regardless of the basic receive channel.
3. UART Buffer Overruns: Modern USB-to-MIDI interfaces fire packets much faster than 1983 Yamaha 6805 CPUs can commit to SRAM. If you send consecutive parameter updates or voice dumps, adding 50ms pacing between packets prevents dropped bytes.

We ran into these exact constraints when building the Web MIDI engine for [bipluk](https://bipluk.com). Once those three checks are in place, the edit buffer updates instantly!
